src/gui/shareusergroupwidget.cpp

```diff
@@ -926,13 +937,54 @@ void ShareUserLine::customizeStyle()
     _deleteShareButton->setIcon(deleteicon);
 
     _ui->noteConfirmButton->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
-    _ui->confirmExpirationDate->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
     _ui->progressIndicator->setColor(QGuiApplication::palette().color(QPalette::WindowText));
 
     // make sure to force BackgroundRole to QPalette::WindowText for a lable, because it's parent always has a different role set that applies to children unless customized
     _ui->errorLabel->setBackgroundRole(QPalette::WindowText);
 }
 
+QPixmap ShareUserLine::pixmapForShareeType(Sharee::Type type, const QColor &backgroundColor) const
+{
+    QString fileName;
+    switch (type) {
+    case Sharee::Room:
+        fileName = QStringLiteral("talk-app.svg");
+        break;
+    case Sharee::Email:
+        fileName = QStringLiteral("email.svg");
+        break;
+    default:
+        return QPixmap();
+    }
+
+    return Theme::instance()->pixmapForBackground(fileName, backgroundColor);
+}
+
+QColor ShareUserLine::backgroundColorForShareeType(Sharee::Type type) const
+{
+    switch (type) {
+    case Sharee::Room:
+        return Theme::instance()->wizardHeaderBackgroundColor();
+    case Sharee::Email:
+        return Theme::instance()->wizardHeaderTitleColor();
+    default:
+        break;
+    }
+
+    const auto calculateBackgroundBasedOnText = [this]() {
+        const auto hash = QCryptographicHash::hash(_ui->sharedWith->text().toUtf8(), QCryptographicHash::Md5);
+        Q_ASSERT(hash.size() > 0);
+        if (hash.size() == 0) {
+            qCWarning(lcSharing) << "Failed to calculate hash color for share:" << _share->path();
+            return QColor{};
```

Comment:
Could be just {}

Reply:
Oh, right it's a lambda. Another reason to use a static function ;)

Reply:
Would it be more verbose? I think it depends. If you have a lot of branches that return the default/empty color {}. Then it makes sense because then you don't repeat yourself too often.

Reply:
> Oh, right it's a lambda. Another reason to use a static function ;)

@FlexW I don't believe being able to return `{}` is a good reason for a drastic change such as turning lambda which is used in one place only into a static function :)